syftr/cli.py

```diff
@@ -0,0 +1,297 @@
+import logging
+import os
+from importlib.metadata import version
+from pathlib import Path
+from typing import List
+
+import click
+import optuna
+from ray.job_submission import JobSubmissionClient
+
+import syftr.scripts.system_check as system_check
+from syftr.api import Study, SyftrUserAPIError
+from syftr.configuration import cfg
+from syftr.optuna_helper import get_study_names
+from syftr.ray import submit
+
+__version__ = version("syftr")
+
+logging.basicConfig(level=logging.INFO, format="%(message)s")
+
+
+@click.group(invoke_without_command=True)
+@click.version_option(__version__, prog_name="syftr")
+@click.pass_context
+def main(ctx):
+    """syftr command‐line interface for running and managing studies."""
+    if ctx.invoked_subcommand is None:
+        click.echo(ctx.get_help())
+        ctx.exit()
+
+
+@main.command()
+def check():
+    """
+    syftr check
+    ---
+    Checks the system for required dependencies and configurations.
+    """
+    system_check.check()
+
+
+@main.command()
+@click.argument(
+    "config_path",
+    type=click.Path(exists=True, dir_okay=False),
+    required=True,
+)
+@click.option(
+    "--follow/--no-follow",
+    default=False,
+    help="Stream logs until the study completes.",
+)
+def run(config_path: str, follow: bool):
+    """
+    syftr run CONFIG_PATH [--follow]
+
+    Launches a study from YAML configuration file at CONFIG_PATH,
+    optionally following the logs until completion.
+    """
+    try:
+        study = Study.from_file(Path(config_path))
+        study.run()
+        if follow:
+            study.wait_for_completion(stream_logs=True)
+    except SyftrUserAPIError as e:
+        click.echo(f"✗ {e}", err=True)
+        raise click.Abort()
+
+
+def _get_ray_job_ids_from_name(client: JobSubmissionClient, job_name: str) -> List[str]:
```

Comment:
```suggestion
def _get_ray_job_ids_from_name(client: JobSubmissionClient, study_name: str) -> List[str]:
```

I think this is more accurate, no? `job_name` is throwing me off